provider/openstack/upgrades.go

```diff
@@ -72,9 +72,19 @@ type tagExistingSecurityGroupsStep struct {
 	env *Environ
 }
 
+var tagSecGroupRetryStrategy = retry.CallArgs{
+	Clock:       clock.WallClock,
+	MaxDuration: 30 * time.Second,
+	Delay:       time.Second,
+	BackoffFunc: retry.ExpBackoff(time.Second, 10*time.Second, 1.5, true),
```

Comment:
Can we tune the parameters and/or max duration a bit.
The back off wait times will be 10, 15, 22.5, ~34 etc.
I think we can set MaxAttempts rather than MaxDuration. Maybe allow for 5 attempts.